functions/src/resolve-market.ts

```diff
@@ -31,140 +35,207 @@ export const resolveMarket = functions
     ) => {
       const userId = context?.auth?.uid
       if (!userId) return { status: 'error', message: 'Not authorized' }
-
-      const { outcome, contractId, probabilityInt, resolutions, value } = data
-
-      const contractDoc = firestore.doc(`contracts/${contractId}`)
+      const contractDoc = firestore.doc(`contracts/${data.contractId}`)
       const contractSnap = await contractDoc.get()
       if (!contractSnap.exists)
         return { status: 'error', message: 'Invalid contract' }
       const contract = contractSnap.data() as Contract
-      const { creatorId, outcomeType, closeTime } = contract
-
-      if (outcomeType === 'BINARY') {
-        if (!RESOLUTIONS.includes(outcome))
-          return { status: 'error', message: 'Invalid outcome' }
-      } else if (outcomeType === 'FREE_RESPONSE') {
-        if (
-          isNaN(+outcome) &&
-          !(outcome === 'MKT' && resolutions) &&
-          outcome !== 'CANCEL'
+      if (contract.creatorId !== userId)
+        return { status: 'error', message: 'User not creator of contract' }
+
+      return privateResolveMarket(contract, data, contractDoc)
+    }
+  )
+
+export const autoResolveMarkets = functions.pubsub
+  .schedule('every 1 hours')
+  .onRun(async () => {
+    const contracts = await getValues<Contract>(
+      firestore.collection('contracts')
+    )
+
+    const contractsToResolve = contracts
+      .filter((c) => !c.isResolved)
+      .filter((c) => c.autoResolutionTime < Date.now())
```

Comment:
Why not query for this? Filtering them in memory seems super gross.

Reply:
Fixed